logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/RegisterClientUseCase.kt

```diff
@@ -177,6 +155,30 @@ class RegisterClientUseCaseImpl @OptIn(DelicateKaliumApi::class) internal constr
             })
     }
 
+    /**
+     * Depending if the build is able to use async notifications and the BE allows it, then add the capability for
+     * [ClientCapability.ConsumableNotifications] otherwise fallback to the current behavior,
+     * just with the [ClientCapability.LegalHoldImplicitConsent] capability.
+     *
+     * When ACK is stable from BE perspective, this can be later moved to the API level, like it was before this change.
+     */
+    private suspend fun RegisterClientParameters.withConsumableNotificationCapabilityWhenAllowed(): RegisterClientParameters {
+        return if (isAllowedToUseAsyncNotifications()) {
+            this.copy(
+                capabilities = capabilities?.toMutableSet()?.apply {
+                    add(ClientCapability.LegalHoldImplicitConsent)
+                    add(ClientCapability.ConsumableNotifications)
+                }?.toList() ?: listOf(ClientCapability.LegalHoldImplicitConsent, ClientCapability.ConsumableNotifications),
+            )
+        } else {
+            this.copy(
+                capabilities = capabilities?.toMutableSet()?.apply {
+                    add(ClientCapability.LegalHoldImplicitConsent)
+                }?.toList() ?: listOf(ClientCapability.LegalHoldImplicitConsent),
+            )
+        }
```

Comment:
That's quite subjective but in my opinion this is more readable 😄 
```suggestion
        this.copy(
            capabilities =  capabilities.orEmpty().toMutableSet().apply {
                add(ClientCapability.LegalHoldImplicitConsent)
                if (isAllowedToUseAsyncNotifications()) {
                    add(ClientCapability.ConsumableNotifications)
                }
            }.toList()
        )
```